packages/smart-contracts/src/contracts/EthConversionProxy.sol

```diff
@@ -0,0 +1,123 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./ChainlinkConversionPath.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+
+/**
+ * @title EthConversionProxy
+ * @notice This contract convert from chainlink then swaps ETH
+ *         before paying a request thanks to a conversion payment proxy
+ *         the dependance with ReentrancyGuard is required to perform
+ *         "transferExactEthWithReferenceAndFee" of the eth-fee-proxy contract
+  */
+contract EthConversionProxy is ReentrancyGuard {
+  address public paymentProxy;
+  ChainlinkConversionPath public chainlinkConversionPath;
+  address public nativeTokenHash;
+
+  constructor(address _paymentProxyAddress, address _chainlinkConversionPathAddress, address _nativeTokenHash) {
+    paymentProxy = _paymentProxyAddress;
+    chainlinkConversionPath = ChainlinkConversionPath(_chainlinkConversionPathAddress);
+    nativeTokenHash = _nativeTokenHash;
+  }
+
+  // Event to declare a conversion with a reference
+  event TransferWithConversionAndReference(
+    uint256 amount,
+    address currency,
+    bytes indexed paymentReference,
+    uint256 feeAmount,
+    uint256 maxRateTimespan
+  );
+  
+  // Event to declare a transfer with a reference
+  event TransferWithReferenceAndFee(
+    address to,
+    uint256 amount,
+    bytes indexed paymentReference,
+    uint256 feeAmount,
+    address feeAddress
+  );
```

Comment:
why is this event re-defined, shouldn't it use the eth-fee-proxy contract?

Reply:
as we do a delegate call the event `TransferWithReferenceAndFee()` will be emit by this this contract.
We declare it to have it in the ABI, making the event fliter either.

Reply:
hum OK, I didn't know this was needed, thanks

Reply:
It's not required but convenient :)

Reply:
could be useful to add a comment?

Reply:
I will then